x/contractmanager/keeper/failure.go

```diff
@@ -0,0 +1,70 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/neutron-org/neutron/x/contractmanager/types"
+)
+
+// AddContractFailure adds a specific failure to the store using address as the key
+func (k Keeper) AddContractFailure(ctx sdk.Context, address string, ackID uint64, ackType string) {
+	failure := types.Failure{
+		Address: address,
+		AckId:   ackID,
+		AckType: ackType,
+	}
+	nextFailureID := k.GetNextFailureIDKey(ctx, failure.GetAddress())
+
+	store := ctx.KVStore(k.storeKey)
+
+	failure.Id = nextFailureID
+	b := k.cdc.MustMarshal(&failure)
+	store.Set(types.GetFailureKey(failure.GetAddress(), nextFailureID), b)
+}
+
+func (k Keeper) GetNextFailureIDKey(ctx sdk.Context, address string) uint64 {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.GetFailureKeyPrefix(address))
+	iterator := sdk.KVStoreReversePrefixIterator(store, []byte{})
+	defer iterator.Close()
+
+	if iterator.Valid() {
+		var val types.Failure
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+
+		return val.Id + 1
+	}
+
+	return 0
```

Comment:
are we sure we want to be so silent about this?

Reply:
Yes, because our first ID is 0. We will return 0 in case if there is no any elements in the store.